pkg/api/cluster_types.go

```diff
@@ -490,18 +490,14 @@ func (cluster *Cluster) SetDynamicCapacityInfo(dynamicCapacityInfo map[string]Dy
 }
 
 // RetrieveDynamicCapacityInfo returns the dynamic scaling info per instance type
-func (cluster *Cluster) RetrieveDynamicCapacityInfo() (map[string]DynamicCapacityInfo, error) {
+func (cluster *Cluster) RetrieveDynamicCapacityInfo() map[string]DynamicCapacityInfo {
 	dynamicCapacityInfo := map[string]DynamicCapacityInfo{}
-	if cluster.DynamicCapacityInfo == nil {
-		return dynamicCapacityInfo, nil
+	if cluster.DynamicCapacityInfo != nil {
+		// ignore error returned by Unmarshal as the json stored in the cluster object will always be a valid DynamicCapacityInfo json object.
+		_ = json.Unmarshal(cluster.DynamicCapacityInfo, &dynamicCapacityInfo)
```

Comment:
I'm ok with the function not returning any error, but I believe we shouldn't ignore JSON errors.
I prefer to log it if an error occurs: it will never happen, but in case we have at least some log about it.

Reply:
makes sense, i'll log it here instead :+1: